File: Examples/VisualBasic/ModifyingAndConvertingImages/DjVu/ConvertRangeOfDjVuPages.vb
Imports Aspose.Imaging.FileFormats.Djvu
Imports Aspose.Imaging.FileFormats.Tiff.Enums
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.DjVu
    Public Class ConvertRangeOfDjVuPages
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_DjVu()

            ' Load a DjVu image
            Using image1 As DjvuImage = DirectCast(Image.Load(dataDir & Convert.ToString("Sample.djvu")), DjvuImage)
                ' Create an instance of TiffOptions with preset options
                Dim exportOptions As New TiffOptions(TiffExpectedFormat.TiffDeflateBw)
                ' Create an instance of IntRange and initialize it with range of pages to be exported
                Dim range As New IntRange(0, 2)
                'Export first 2 pages
                ' Initialize an instance of DjvuMultiPageOptions while passing instance of IntRange
                exportOptions.MultiPageOptions = New DjvuMultiPageOptions(range)
                ' Call Save method while passing instance of TiffOptions
                image1.Save(dataDir & Convert.ToString("ConvertRangeOfDjVuPages_out.djvu"), exportOptions)
            End Using
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/DjVu/ConvertRangeOfDjVuPages.cs
// GIST-ID: cec93d0d381642fe9f74340f6f8bbc1b
using Aspose.Imaging.FileFormats.Djvu;
using Aspose.Imaging.FileFormats.Tiff.Enums;
using Aspose.Imaging.ImageOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.DjVu
{
    public class ConvertRangeOfDjVuPages
    {
        public static void Run()
        {
            Console.WriteLine("Running example ConvertRangeOfDjVuPages");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_DjVu();

            // Load a DjVu image
            using (DjvuImage image = (DjvuImage)Image.Load(dataDir + "Sample.djvu"))
            {
                // Create an instance of TiffOptions with preset options and an IntRange, and initialize it with the range of pages to be exported
                TiffOptions exportOptions = new TiffOptions(TiffExpectedFormat.TiffDeflateBw);
                IntRange range = new IntRange(0, 2);

                // Initialize an instance of DjvuMultiPageOptions by passing the IntRange, and call Save method passing the TiffOptions
                exportOptions.MultiPageOptions = new DjvuMultiPageOptions(range);
                image.Save(dataDir + "ConvertRangeOfDjVuPages_out.djvu", exportOptions);
            }

            Console.WriteLine("Finished example ConvertRangeOfDjVuPages");
        }
    }
}